### PR TITLE
docs: refactor contributing guidelines - add docs guidelines

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

Based off [Cosmos-SDK's](https://github.com/cosmos/cosmos-sdk/blob/main/docs/DOC_WRITING_GUIDELINES.md).

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]
Beautiful ❤️

File: docs/DOCS_GUIDELINES.MD [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Documentation guidelines
+
+## Best practices
+
+- Check the spelling and grammar, even if you have to copy and paste from an external source.
+- Use simple sentences. Easy-to-read sentences mean the reader can quickly use the guidance you share.
+- Try to express your thoughts in a concise and clean way.
+- Leave a space between the acronyms ADR and ICS and the corresponding number (e.g. ADR 008 and ICS 27).
+- Don't abuse `code` format when writing in plain English.
```
Comment (colin-axner) [inline docs/DOCS_GUIDELINES.MD:9]: What does this mean? Also I find `abuse` to be strong language. Maybe `overuse` is better?

**Review by chatton — APPROVED** [review]
LGTM!
